### PR TITLE
Show user debug details in debug info

### PR DESCRIPTION
**Pull request by partydragen**
This allow us to validate if the user has the groups example if we're investigating their permissions problems to view form submissions example

### CODE REVIEW

**Comment by partydragen** [comment]
```
    "user": {
        "id": 1,
        "username": "Partydragen",
        "nickname": "Partydragen",
        "groups": [
            2,
            8,
            7
        ],
        "integrations": {
            "Patreon": {
                "username": "Partydragen",
                "identifier": "5206074",
                "verified": true
            },
            "MC-Server-List": {
                "username": "Partydragen",
                "identifier": "1",
                "verified": true
            },
            "CubedCraft": {
                "username": "Partydragen",
                "identifier": "1",
                "verified": true
            },
            "MCStatistics": {
                "username": "Partydragen",
                "identifier": "1",
                "verified": true
            },
            "Discord": {
                "username": "Partydragen#0001",
                "identifier": "117749257569566723",
                "verified": true
            }
        }
    },
```

**Comment by samerton** [comment]
Would it be more useful to return
```json
"groups": [
  {
    "id": 2,
    "name": "Admin"
  }
]
```
or similar?

**Comment by partydragen** [comment]
> Would it be more useful to return
> 
> ```json
> "groups": [
>   {
>     "id": 2,
>     "name": "Admin"
>   }
> ]
> ```
> 
> or similar?

The debug info already have the group list, so the fancy debug page can already collect the group info from that list like several other things work there

File: modules/Core/queries/debug_link.php [hunk 1]
```diff
@@ -179,10 +179,36 @@
     $logs[$type] = file_exists($file_path) ? Util::readFileEnd($file_path, $max_bytes = 10_000) : '';
 }
 
+$user_data = [];
+if ($user->isLoggedIn()) {
+    $user_groups = [];
+    foreach ($user->getAllGroupIds() as $group_id) {
+        $user_groups[] = $group_id;
```
Comment (tadhgboyle) [inline modules/Core/queries/debug_link.php]: Is this any different than doing "$user_groups = $user->getAllGroupIds();"?

File: modules/Core/queries/debug_link.php [hunk 2]
```diff
@@ -179,10 +179,36 @@
...
+    }
+
+    $user_integrations = [];
+    foreach ($user->getIntegrations() as $integrationUser) {
+        $user_integrations[$integrationUser->getIntegration()->getName()] = [
+            'username' => $integrationUser->data()->username,
+            'identifier' => $integrationUser->data()->identifier,
+            'verified' => $integrationUser->data()->verified ? true : false
```
Comment (tadhgboyle) [inline modules/Core/queries/debug_link.php]: We probably don't need this ternary if "verified" returns a bool already

File: modules/Core/queries/debug_link.php [hunk 3]
```diff
@@ -179,10 +179,36 @@
...
+        'groups' => $user_groups,
+        'integrations' => $user_integrations
+    ];
+}
+
 $data = [
     'generated_at' => time(),
     'generated_by_name' => $user->data()->username,
     'generated_by_uuid' => $uuid,
+    "user" => $user_data,
```
Comment (tadhgboyle) [inline modules/Core/queries/debug_link.php]: Single quotes!

**Review by tadhgboyle — CHANGES_REQUESTED** [review]

File: modules/Core/queries/debug_link.php (hunk 1 above)
Comment (partydragen) [reply modules/Core/queries/debug_link.php]: that returned stupid 8 => 8, 3 => 3 etc

Comment (tadhgboyle) [reply modules/Core/queries/debug_link.php]: array_values($user->getAllGroupIds());

Comment (partydragen) [reply modules/Core/queries/debug_link.php]: That worked :D

File: modules/Core/queries/debug_link.php (hunk 2 above)
Comment (partydragen) [reply modules/Core/queries/debug_link.php]: Solved

File: modules/Core/queries/debug_link.php (hunk 3 above)
Comment (partydragen) [reply modules/Core/queries/debug_link.php]: Solved

**Review by tadhgboyle — APPROVED** [review]

**Pull request merged**